apps/scripts/Matrix_Mul.c

```diff
@@ -0,0 +1,50 @@
+void multiply(int mat1[][3], int mat2[][3], int res[][3])
+{
+    int i, j, k;
+    for (i = 0; i < 3; i++) {
+        for (j = 0; j < 3; j++) {
+            res[i][j] = 0;
+            for (k = 0; k < 3; k++)
+                res[i][j] += mat1[i][k] * mat2[k][j];
+        }
+    }
+}
+ #define MMIO_GENERAL  (*(volatile int (*)[64])(0x00400F00))//
+ 
+int main()
+{
+    int mat1[3][3];              
+    mat1[0][0] = 1;
+    mat1[0][1] = 2;
+    mat1[0][2] = 3;
+    mat1[1][0] = 4;
+    mat1[1][1] = 5;
+    mat1[1][2] = 6;
+    mat1[2][0] = 7;
+    mat1[2][1] = 8;
+    mat1[2][2] = 9;
+    int mat2[3][3];
+    mat2[0][0] = 9;
+    mat2[0][1] = 8;
+    mat2[0][2] = 7;
+    mat2[1][0] = 6;
+    mat2[1][1] = 5;
+    mat2[1][2] = 4;
+    mat2[2][0] = 3;
+    mat2[2][1] = 2;
+    mat2[2][2] = 1;
+    int res[3][3]; // To store result
+    multiply(mat1, mat2, res);
+    MMIO_GENERAL[0] = res[0][0];
```

Comment:
Perhaps you could use a loop here (or use/implement some memcpy-like function)

Reply:
this was done only so i could see a sequential write to memory in logs
after creating a log to shared mem access i can now change this to loop